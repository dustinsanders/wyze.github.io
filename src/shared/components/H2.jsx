// @flow

import type { Style, VNode } from '../../types'
import { css } from 'glamor'
import preact from 'preact'

type Props = {
  children?: VNode,
  styles?: Style,
}

const h2: Style = css({
  fontSize: '1.5em',
})

const H2 = ({ children, styles }: Props): VNode => (
  <h2 {...css(h2, styles)}>{children}</h2>
)

H2.defaultProps = {
  children: null,
  styles: {},
}

export default H2
