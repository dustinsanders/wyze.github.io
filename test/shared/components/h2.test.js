// @flow

import H2 from '../../../src/shared/components/H2'
import preact from 'preact'
import snapshot from '../../helpers/snapshot'

const children = (
  // eslint-disable-next-line react/jsx-filename-extension
  <span>Hello</span>
)

const styles = {
  color: 'rebeccapurple',
}

describe('<H2 />', () => {
  it('renders', snapshot.bind(null, H2, { children }))
  it('renders styles', snapshot.bind(null, H2, { children, styles }))
})
