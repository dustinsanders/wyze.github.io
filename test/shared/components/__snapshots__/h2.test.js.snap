// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H2 /> renders 1`] = `
"<h2 data-css-tc8z0v>
  <span>Hello</span>
</h2>"
`;

exports[`<H2 /> renders styles 1`] = `
"<h2 data-css-f865j5>
  <span>Hello</span>
</h2>"
`;
